gcc/rust/ast/rust-path.h

```diff
@@ -736,10 +745,13 @@ class PathInExpression : public Pattern, public ExprWithoutBlock
   bool is_error () const
   {
     // FIXME: Cleanup
-    if (path->get_path_kind () == Path::Kind::Regular)
-      return !static_cast<RegularPath &> (*path).has_segments ();
-
-    return false;
+    switch (path->get_path_kind ())
+      {
+      case Path::Kind::Regular:
+	return !static_cast<RegularPath &> (*path).has_segments ();
+      default:
+	rust_unreachable ();
+      }
```

Comment:
actually the old version is better, but we need to turn the `return false` into a `rust_unreachable`